### ISSUE TITLE
sorting mentee list prior to matching

### ISSUE THREAD

**Issue by ageller** [CIERA-Northwestern/JEDI_ext_mentorship#11]
Should we sort by the number of mentors a given mentee already has?

**Comment by TStarkenburg** [comment]
We're thinking of splitting by this. If someone has more mentors than others at the beginning of a round (for example because of direct matching), they will not be considered as mentee in that round. and only included again when others have the same number of mentors. 
This gets a bit tricky though, because people will have less mentors if they cannot be matched. After not being able to be matched once, its ok to first try to match these again before adding mentors for others, but you don't want to get in an infinite loop this way....

**Comment by agurvich** [comment]
See PR https://github.com/CIERA-Northwestern/JEDI_ext_mentorship/pull/16. It is not quite as we discussed before as I have it now but I can change it if we decide we want to. 

The difference is that rather than ruling a mentee with 3 mentors ineligible for random matching until all other mentees have 3 (which may not be possible based on mentor/mentee avoids) I have it so that they are ruled ineligible only for the first 3 rounds. I think this addresses the spirit of the idea and prevents us from getting deadlocked. 

I also removed the infinite loop and instead only do `max(n_mentors_total)+5` rounds (the +5 is in case someone got unlucky and couldn't be matched, which I'm not totally sure can happen). I kept the break statement but instead check if either list is empty.
